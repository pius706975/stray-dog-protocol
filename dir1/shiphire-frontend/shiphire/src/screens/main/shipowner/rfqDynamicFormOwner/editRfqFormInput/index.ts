export { default as EditRFQInputFormOwner } from './EditRFQInputFormOwner';
