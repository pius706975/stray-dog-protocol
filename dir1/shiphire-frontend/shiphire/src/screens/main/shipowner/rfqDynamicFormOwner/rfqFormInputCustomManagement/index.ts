export { default as RFQFormInputCustomManagementOwner } from './RFQFormInputCustomManagementOwner';
