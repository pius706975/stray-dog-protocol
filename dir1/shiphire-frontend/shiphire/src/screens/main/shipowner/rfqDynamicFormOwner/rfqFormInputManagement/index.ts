export { default as RFQFormInputManagementOwner } from './RFQFormInputManagementOwner';
