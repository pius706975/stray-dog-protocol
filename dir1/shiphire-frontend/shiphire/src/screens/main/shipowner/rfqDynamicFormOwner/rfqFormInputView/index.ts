export { default as RFQFormInputView } from './RFQFormInputView';
