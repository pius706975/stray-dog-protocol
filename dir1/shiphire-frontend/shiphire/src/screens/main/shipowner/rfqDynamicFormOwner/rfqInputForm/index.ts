export { default as RFQInputFormOwner } from './RFQInputFormOwner';
