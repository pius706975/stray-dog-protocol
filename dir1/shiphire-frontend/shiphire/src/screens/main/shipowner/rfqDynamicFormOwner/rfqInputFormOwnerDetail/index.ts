export { default as RFQInputFormOwnerDetail } from './RFQInputFormOwnerDetail';
