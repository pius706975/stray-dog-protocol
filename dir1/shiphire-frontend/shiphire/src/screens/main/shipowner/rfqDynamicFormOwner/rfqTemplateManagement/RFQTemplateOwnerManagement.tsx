import React from 'react';
import { Pressable } from 'react-native';
import { View } from 'react-native-ui-lib';
import { CustomText } from '../../../../../components';
import { Color } from '../../../../../configs';
import {
    DynamicFormRFQ,
    RFQTemplateManagementOwnerProps,
} from '../../../../../types';
import {
    createTemplateCustomRFQForm,
    useGetTemplateRfqShipFormByShipCategory,
} from '../../../../../hooks';
import { useIsFocused } from '@react-navigation/native';
import { useDispatch } from 'react-redux';
import { modalSlice, progressIndicatorSlice } from '../../../../../slices';
import { handleAxiosError } from '../../../../../utils';
import { useTranslation } from 'react-i18next';

const RFQTemplateOwnerManagement: React.FC<RFQTemplateManagementOwnerProps> = ({
    navigation,
    route,
}) => {
    const { shipCategory, shipId } = route.params;
    const { hideProgressIndicator, showProgressIndicator } =
        progressIndicatorSlice.actions;
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const mutationGetTemplateShipRFQForm =
        useGetTemplateRfqShipFormByShipCategory();
    const [dynamicFormData, setDynamicFormData] = React.useState<
        DynamicFormRFQ | undefined
    >();
    const mutationCreateTemplateCustomRFQ = createTemplateCustomRFQForm();
    const { t } = useTranslation('rfq');
    const isFocused = useIsFocused();
    React.useEffect(() => {
        isFocused &&
            mutationGetTemplateShipRFQForm.mutate(shipCategory, {
                onSuccess: resp => {
                    setDynamicFormData(resp.data.data);
                },
                onError: err => {
                    console.log(err);
                },
            });
    }, [isFocused]);

    const customRFQ = () => {
        dispatch(showProgressIndicator());

        mutationCreateTemplateCustomRFQ.mutate(
            { shipId },
            {
                onSuccess: () => {
                    dispatch(hideProgressIndicator());
                    dispatch(
                        showModal({
                            status: 'success',
                            text: t('RFQDynamicForm.SuccessCreatedCustomRFQ'),
                        }),
                    );

                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 3000);

                    navigation.navigate('RFQFormInputCustomManagementOwner', {
                        shipId: shipId,
                        formType: dynamicFormData?.formType,
                        templateType: dynamicFormData?.templateType,
                    });
                },
                onError: err => {
                    dispatch(hideProgressIndicator());
                    dispatch(
                        showModal({
                            status: 'failed',
                            text: 'Add RFQ Template Failed/Duplicate Template',
                        }),
                    );

                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 2000);
                    handleAxiosError(err);
                },
            },
        );
    };
    console.log(dynamicFormData);

    return (
        <View style={{ flex: 1 }}>
            {dynamicFormData ? (
                <View
                    style={{
                        padding: 10,
                    }}>
                    <Pressable
                        onPress={() => {
                            navigation.navigate('RFQFormInputManagementOwner', {
                                dynamicForms: dynamicFormData?.dynamicForms,
                                shipId: shipId,
                                templateType: dynamicFormData?.templateType,
                            });
                        }}>
                        <View
                            style={{
                                borderBottomWidth: 0.5,
                                borderBottomColor: Color.darkTextColor,
                                paddingVertical: 10,
                                paddingHorizontal: 20,
                            }}>
                            <CustomText
                                fontFamily="regular"
                                fontSize="lg"
                                color="darkTextColor">
                                {dynamicFormData && shipCategory}{' '}
                                {t('RFQDynamicForm.DefaultRFQTemplatesText')}
                            </CustomText>
                        </View>
                    </Pressable>
                    <Pressable testID="custom-rfq" onPress={customRFQ}>
                        <View
                            style={{
                                borderBottomWidth: 0.5,
                                borderBottomColor: Color.darkTextColor,
                                paddingVertical: 10,
                                paddingHorizontal: 20,
                            }}>
                            <CustomText
                                fontFamily="regular"
                                fontSize="lg"
                                color="darkTextColor">
                                {t('RFQDynamicForm.CustomRFQTemplatesText')}
                            </CustomText>
                        </View>
                    </Pressable>
                </View>
            ) : (
                <View
                    style={{
                        flex: 1,
                        justifyContent: 'center',
                        alignItems: 'center',
                    }}>
                    <CustomText
                        color="darkTextColor"
                        fontSize="lg"
                        fontFamily="regular">
                        {t('RFQDynamicForm.EmptyRFQTemplatesText')}
                    </CustomText>
                </View>
            )}
        </View>
    );
};

export default RFQTemplateOwnerManagement;
