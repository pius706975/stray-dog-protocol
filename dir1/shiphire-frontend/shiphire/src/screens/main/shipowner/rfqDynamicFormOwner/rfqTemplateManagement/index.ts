export { default as RFQTemplateOwnerManagement } from './RFQTemplateOwnerManagement';
