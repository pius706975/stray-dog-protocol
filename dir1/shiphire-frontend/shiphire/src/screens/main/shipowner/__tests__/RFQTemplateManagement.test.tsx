import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { MainOwnerStackParamList } from '../../../../types';
import { QueryClient, QueryClientProvider } from 'react-query';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import { RFQTemplateOwnerManagement } from '../rfqDynamicFormOwner/rfqTemplateManagement';
import {
    createTemplateCustomRFQForm,
    useGetTemplateRfqShipFormByShipCategory,
} from '../../../../hooks';
import { fireEvent, render } from '@testing-library/react-native';

jest.mock('../../../../hooks');
const MainOwnerStack = createNativeStackNavigator<MainOwnerStackParamList>();
const queryClient = new QueryClient();
const mockAdapter = new MockAdapter(httpRequest);

const mockParams = {
    shipCategory: '123',
    shipId: '234',
};

const ownerStackMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwnerStack.Navigator>
                    <MainOwnerStack.Screen
                        name="RFQTemplateOwnerManagement"
                        component={RFQTemplateOwnerManagement}
                        initialParams={mockParams}
                    />
                </MainOwnerStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

const mockCreateTemplate = () => {
    mockAdapter.onPost('/admin/save-custom-template-RFQ').reply(200, {
        message: 'success',
    });
    (createTemplateCustomRFQForm as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                message: 'success',
            };
            onSuccess(mockResponseData);
        }),
    }));
};
const mockCreateTemplateError = () => {
    mockAdapter.onPost('/admin/save-custom-template-RFQ').reply(500, {
        message: 'failed',
    });
    (createTemplateCustomRFQForm as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onError } = options;
            const mockResponseData = {
                message: 'failed',
            };
            onError(mockResponseData);
        }),
    }));
};

const mockTemplateData = {
    _id: '65af73fda6a0ba4f3492eb34',
    formType: 'rfqForm',
    templateType: 'Self Propelled Oil BargeRfq',
    dynamicForms: [
        {
            dynamicInput: {
                _id: '65a737c476517457909c5f42',
                formType: 'rfqForm',
                templateType: 'defaultRfq',
                inputType: 'textInput',
                label: 'Needs',
                fieldName: 'needs',
                fieldType: 'string',
                placeholder: 'Ex: for transporting',
                active: true,
                __v: 0,
            },
            required: true,
            option: [],
            _id: '65af73fda6a0ba4f3492eb35',
        },
        {
            dynamicInput: {
                _id: '65a737c476517457909c5f43',
                formType: 'rfqForm',
                templateType: 'defaultRfq',
                inputType: 'radioDropdown',
                label: 'Ship Rent Type',
                fieldName: 'shipRentType',
                fieldType: 'string',
                placeholder: 'Select your ship rent type',
                active: true,
                __v: 0,
            },
            required: true,
            option: [],
            _id: '65af73fda6a0ba4f3492eb36',
        },
        {
            dynamicInput: {
                _id: '65a737c476517457909c5f44',
                formType: 'rfqForm',
                templateType: 'defaultRfq',
                inputType: 'textInput',
                label: 'Location Departure',
                fieldName: 'locationDeparture',
                fieldType: 'string',
                placeholder: 'Ex: Palu, Sulawesi Tengah',
                active: true,
                __v: 0,
            },
            required: true,
            option: [],
            _id: '65af73fda6a0ba4f3492eb37',
        },
        {
            dynamicInput: {
                _id: '65a737c476517457909c5f45',
                formType: 'rfqForm',
                templateType: 'defaultRfq',
                inputType: 'textInput',
                label: 'Location Destination',
                fieldName: 'locationDestination',
                fieldType: 'string',
                placeholder: 'Ex: Surabaya, Jawa Timur',
                active: true,
                __v: 0,
            },
            required: true,
            option: [],
            _id: '65af73fda6a0ba4f3492eb38',
        },
        {
            dynamicInput: {
                _id: '65a737c476517457909c5f46',
                formType: 'rfqForm',
                templateType: 'defaultRfq',
                inputType: 'datePickerCalendar',
                label: 'Rental Date',
                fieldName: 'rentalDate',
                fieldType: 'string',
                placeholder: 'Select date',
                active: true,
                __v: 0,
            },
            required: true,
            option: [],
            _id: '65af73fda6a0ba4f3492eb39',
        },
        {
            dynamicInput: {
                _id: '65a737c476517457909c5f47',
                formType: 'rfqForm',
                templateType: 'defaultRfq',
                inputType: 'textInput',
                label: 'Additional Information',
                fieldName: 'additionalInformation',
                fieldType: 'string',
                placeholder: 'Ex: Additioal Crew, etc',
                active: true,
                __v: 0,
            },
            required: true,
            option: [],
            _id: '65af73fda6a0ba4f3492eb3a',
        },
    ],
    active: true,
    createdAt: '2024-01-23T08:08:29.211Z',
    updatedAt: '2024-01-23T08:08:29.211Z',
    __v: 0,
};

beforeEach(() => {
    mockAdapter.onGet('/admin/get-dynamic-input').reply(200, {
        message: 'success',
        data: mockTemplateData,
    });
    (useGetTemplateRfqShipFormByShipCategory as jest.Mock).mockImplementation(
        () => ({
            mutate: jest.fn().mockImplementation((data, options) => {
                const { onSuccess } = options;
                const mockResponseData = {
                    data: {
                        data: mockTemplateData,
                    },
                };
                onSuccess(mockResponseData);
            }),
        }),
    );
});

describe('RFQ Template Management', () => {
    describe('Snapshot Testing', () => {
        it('should render RFQ Template Management screen correctly', () => {
            const templateScreen = render(ownerStackMockComponent);
            expect(templateScreen).toMatchSnapshot();
        });
    });

    describe('Unit Testing', () => {
        it('should save custom RFQ', () => {
            mockCreateTemplate();
            const { getByTestId } = render(ownerStackMockComponent);
            const customPressable = getByTestId('custom-rfq');
            fireEvent.press(customPressable);
            expect(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'success',
                    text: 'RFQDynamicForm.SuccessCreatedCustomRFQ',
                    visible: true,
                });
            });
        });
        it('should show modal when failed save custom RFQ', () => {
            mockCreateTemplateError();
            const { getByTestId } = render(ownerStackMockComponent);
            const customPressable = getByTestId('custom-rfq');
            fireEvent.press(customPressable);
            expect(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'failed',
                    text: 'Add RFQ Template Failed/Duplicate Template',
                    visible: true,
                });
            });
        });
    });
});
