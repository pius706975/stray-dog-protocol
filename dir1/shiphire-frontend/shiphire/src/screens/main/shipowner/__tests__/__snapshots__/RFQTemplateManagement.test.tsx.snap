// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RFQ Template Management Snapshot Testing should render RFQ Template Management screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              style={
                [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  {
                    "padding": 10,
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "borderBottomColor": "#4d5963",
                        "borderBottomWidth": 0.5,
                        "paddingHorizontal": 20,
                        "paddingVertical": 10,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#4d5963",
                        "fontFamily": "Kanit-Regular",
                        "fontSize": 20,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    123
                     
                    RFQDynamicForm.DefaultRFQTemplatesText
                  </Text>
                </View>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                testID="custom-rfq"
              >
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "borderBottomColor": "#4d5963",
                        "borderBottomWidth": 0.5,
                        "paddingHorizontal": 20,
                        "paddingVertical": 10,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#4d5963",
                        "fontFamily": "Kanit-Regular",
                        "fontSize": 20,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    RFQDynamicForm.CustomRFQTemplatesText
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="RFQTemplateOwnerManagement"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
